src/netbase.cpp

```diff
@@ -212,6 +212,24 @@ CService LookupNumeric(const std::string& name, uint16_t portDefault, DNSLookupF
     return Lookup(name, portDefault, /*fAllowLookup=*/false, dns_lookup_function).value_or(CService{});
 }
 
+bool IsUnixSocketPath(const std::string& name)
+{
+#if HAVE_SOCKADDR_UN
+    if (name.find(ADDR_PREFIX_UNIX) != 0) return false;
+
+    // Split off "unix:" prefix
+    std::string str{name.substr(ADDR_PREFIX_UNIX.length())};
+
+    // Path size limit is platform-dependent
+    // see https://manpages.ubuntu.com/manpages/xenial/en/man7/unix.7.html
+    if (str.size() + 1 > sizeof(((sockaddr_un*)nullptr)->sun_path)) return false;
+
+    return true;
+#else
```

Comment:
Hm not sure what you mean by unreliable. Wont the pre-processor directives here guaruntee this function always returns false on systems that don't support unix sockets?

Reply:
It would, but "unix:/somepath" is valid unix socket path, so `IsUnixSocketPath` should return `true`. So it's seems better to ensure nobody accidentally calls this function on a system that doesn't support unix sockets.

Reply:
> is valid unix socket path, 

This can actually only be determined if `sockaddr_un` is defined, because different platforms have different length limits for the path (https://manpages.ubuntu.com/manpages/xenial/en/man7/unix.7.html)

So if `sockaddr_un` is not defined, nothing can be a valid unix socket path.

Also I think `Assume(false)` is overkill because this function will be called from `AppInitMain()` regardless of platform. If a user without unix socket support still tries to set `-onion=unix:/whatever` they'll get an init error